### PR TITLE
G0 5073: On GST transaction page provide filter by adjustment of advance for gstr-1 and invoice management

### PR DESCRIPTION
**Pull request by rohit-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Applied filter adjustment of advance for gstr-1
Invoice management advance receipts filter applied

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts [hunk 1]
```diff
@@ -221,6 +223,18 @@ export class InvoicePreviewComponent implements OnInit, OnChanges, OnDestroy {
     /** Pagination limit */
     public paginationLimit: number = PAGINATION_LIMIT;
     public purchaseRecord: any = {};
+    /**Adjust advance receipts */
+    @ViewChild('adjustPaymentModal') public adjustPaymentModal: ModalDirective;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts]: Add docs for all members declared and avoid introduction of blank lines

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts [hunk 2]
```diff
@@ -601,6 +618,16 @@ export class InvoicePreviewComponent implements OnInit, OnChanges, OnDestroy {
                     this.parseBalRes(res);
                 });
             });
+
+        this.voucherDetails$.subscribe(response => {
+            if (response) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts]: Add nullity check to response.subTotal or based on status 'success'

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts [hunk 3]
```diff
@@ -1408,4 +1434,61 @@ public toggleBulkUpdatePopup(isClose: boolean): void {
         return item;
     }
 
+    /**
+     * To adjustinvoice with advance receipts
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts]: Space between words adjustinvoice

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts [hunk 4]
```diff
@@ -1408,4 +1434,61 @@ public toggleBulkUpdatePopup(isClose: boolean): void {
...
+        this.store.dispatch(this.invoiceReceiptActions.ResetVoucherDetails());
+        // To get re-assign receipts voucher store
+        this.store.dispatch(this.invoiceReceiptActions.getVoucherDetailsV4(item.account.uniqueName, {
+            invoiceNumber: item.voucherNumber,
+            voucherType: VoucherTypeEnum.sales
+        }));
+    }
+
+    /**
+        * To close advance reciipt modal
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts]: Correct formatting of JS doc and reciipt spelling

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts (hunk 2 above)
Comment (rohit-walkover) [reply apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts]: there is no any success status in response only voucher object is there in response

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts (hunk 4 above)
Comment (rohit-walkover) [reply apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts]: done

**Pull request merged**